### PR TITLE
Allowed to clear groups and category after being set

### PR DESCRIPTION
**Pull request by cabidop**
This allows to clear both category and groups fields by enforcing JSON data to be sent to the backend (as by default AJAX calls remove empty arrays from their payload) and by allowing the backend to set fields with an empty/null value.

### CODE REVIEW

File: config/locales/client.en.yml [hunk 1]
```diff
@@ -29,13 +29,14 @@ en:
...
             instructions: "Theme with page assets"
-            select: "Select theme"
+            select: "Select theme…"
           groups:
             label: "Groups"
             instructions: "Limit access to group(s)"
           category:
             label: "Category"
             instructions: "Connect page to category (requires child pages)"
+            select: "Select category…"
```
Comment (cabidop) [inline config/locales/client.en.yml:39]: @angusmcleod I added here a new key to display when no category is selected, but I am not familiar with the i18n process in this repository. Should I use Crowdin (as there is a `crowdin.yml` configuration file) or is there any other method to align the rest of the translations? Should I just remove the other translation files as they are essentially copies of the English (default) one?

**Comment by angusmcleod** [comment]
@cabidop This is great! As a general rule though, please don't do refactoring of code unless it's necessary for the purposes of this specific PR. A PR with other unrelated formatting changes is hard to review as you have to figure out what is relevant and what is not. 

Generally speaking, we follow the same contribution guidelines as Discourse for our Discourse work. Check out those here: https://meta.discourse.org/t/discourse-development-contribution-guidelines/3823.

If you could make this PR about just the thing you've mentioned in the OP I'd be happy to review and merge. If you want to make other formatting changes, please do them together in a separate PR just about formatting.

**Comment by cabidop** [comment]
Hi, @angusmcleod ! I am actually trying to stick to the Discourse contribution guidelines myself too, and it was precisely because of the [refactoring only NOT welcome](https://meta.discourse.org/t/discourse-development-contributor-guidelines/3823#refactoring-only-not-welcome-thumbsdown-19) rule (that encourages this kind of small refactor tied to a feature/fix) that I created the PR.

Happy to move them to a different one in any case, besides the translation the only other "cosmetic" code is the `ComboBox/GroupChooser` update and the `self.` usage in Ruby, the rest is strictly used for the PR.

**Comment by angusmcleod** [comment]
@cabidop Yeah, the issue is that the refactoring is unrelated to the purpose of the PR. That's what makes it difficult for a reviewer to easily understand what the code changes are trying to do. Each PR should be about one thing.

**Comment by angusmcleod** [comment]
@cabidop Thanks 👍

**Pull request merged**